vault/resource_database_secret_backend_role_test.go

```diff
@@ -136,6 +150,21 @@ resource "vault_database_secret_backend_role" "test" {
   max_ttl = 7200
   creation_statements = ["SELECT 1;"]
 }
+
+resource "vault_database_secret_backend_role" "role" {
+  backend             = vault_mount.db.path
+  name                = "dev"
+  db_name             = vault_database_secret_backend_connection.test.name
+  creation_statements = ["{'database_name': '$external', 'x509Type': 'CUSTOMER', 'roles': [{'databaseName':'dbName','roleName':'read'}]}"]
+  credential_config = {
```

Comment:
I think this `credential_config` is only valid for a `credential_type` of `client_certificate` but we aren't setting `credential_type` so the default is being used.

Reply:
Yes, you are right. It was hard to set up the test with the cert so we decided to have a local test for MongoDB instead.